buildSrc/src/main/java/org/apache/kafka/message/HeaderGenerator.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.message;
+
+import java.util.TreeSet;
+
+/**
+ * The Kafka header generator.
+ */
+public final class HeaderGenerator {
+    private static final String[] HEADER = new String[] {
+        "/*",
+        " * Licensed to the Apache Software Foundation (ASF) under one or more",
+        " * contributor license agreements. See the NOTICE file distributed with",
+        " * this work for additional information regarding copyright ownership.",
+        " * The ASF licenses this file to You under the Apache License, Version 2.0",
+        " * (the \"License\"); you may not use this file except in compliance with",
+        " * the License. You may obtain a copy of the License at",
+        " *",
+        " *    http://www.apache.org/licenses/LICENSE-2.0",
+        " *",
+        " * Unless required by applicable law or agreed to in writing, software",
+        " * distributed under the License is distributed on an \"AS IS\" BASIS,",
+        " * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.",
+        " * See the License for the specific language governing permissions and",
+        " * limitations under the License.",
+        " */",
+        "",
+        "// THIS CODE IS AUTOMATICALLY GENERATED.  DO NOT EDIT.",
+        ""
+    };
+
+    static final String PACKAGE = "org.apache.kafka.message";
```

Comment:
That makes sense.  I think we should have:
`org.apache.kafka.common.protocol`: Message, Readable, Writable, MessageUtil
`org.apache.kafka.message`: the generator code
`org.apache.kafka.common.message`: the generated message code

One of the reasons behind the split is that in newer versions of Java, you can only have one directory of sources that provides a certain Java module.

Another reason is because it makes it more clear what is providing what.  There is really no reason to have these three items in the same Java package since they provide distinct things.

Reply:
The suggested packages sound good to me.